src/Microsoft.AspNetCore.ResponseCaching/Internal/NoopResponseCachingCacheKeySuffixProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    public class NoopResponseCachingCacheKeySuffixProvider : IResponseCachingCacheKeySuffixProvider
+    {
+        public string CreateCustomKeySuffix(HttpContext httpContext) => null;
```

Comment:
I think during one of our meetings (or maybe just a offline discussion) we decided that we wanted to always keep the default key and only allow users to append their custom strings as a customized varyby. If we implement the default behaviour here, we give the user complete control over the cache key, is that what we want?